src/wp-includes/block-patterns.php

```diff
@@ -350,6 +350,98 @@ function _register_remote_theme_patterns() {
  * @access private
  */
 function _register_theme_block_patterns() {
+	/*
+	 * Register patterns for the active theme. If the theme is a child theme,
+	 * let it override any patterns from the parent theme that shares the same slug.
+	 */
+	$themes   = array();
+	$theme    = wp_get_theme();
+	$themes[] = $theme;
+	if ( $theme->parent() ) {
+		$themes[] = $theme->parent();
+	}
+	$registry = WP_Block_Patterns_Registry::get_instance();
+
+	foreach ( $themes as $theme ) {
+		$pattern_data = _wp_get_block_patterns( $theme );
+		$dirpath      = $theme->get_stylesheet_directory() . '/patterns/';
+
+		foreach ( $pattern_data['patterns'] as $file => $pattern_data ) {
+			if ( $registry->is_registered( $pattern_data['slug'] ) ) {
+				continue;
+			}
+
+			// The actual pattern content is the output of the file.
+			ob_start();
+			include $dirpath . $file;
+			$pattern_data['content'] = ob_get_clean();
+			if ( ! $pattern_data['content'] ) {
+				continue;
+			}
```

Comment:
Is is not saves in cache by design. This files might be large and we don't want to save that data in a cache.